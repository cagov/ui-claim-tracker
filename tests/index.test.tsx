import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import Index from '../pages/index'

import { useRouter } from 'next/router'

jest.mock('next/router', () => ({
  __esModule: true,
  useRouter: jest.fn(),
}))

describe('Exemplar react-test-renderer Snapshot test', () => {
  it('renders homepage unchanged', () => {
    const mockRouter = {
      locale: 'en',
    }
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    const tree = renderer.create(<Index />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Example react testing-library Test', () => {
  it('has our placeholder app', () => {
    render(<Index />)
    expect(screen.queryByText('UI Online')).toBeInTheDocument()
  })
})
