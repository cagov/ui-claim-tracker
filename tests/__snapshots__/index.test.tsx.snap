// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exemplar react-test-renderer Snapshot test renders homepage unchanged 1`] = `
<div
  className="index container-fluid"
>
  <main
    className="main"
  >
    <header
      className="header border-bottom border-secondary"
    >
      <nav
        className="justify-content-between navbar navbar-expand-lg navbar-dark"
        fixed-top="true"
      >
        <a
          className="navbar-brand"
          href="https://ca.gov"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            height="34"
            src="/images/Ca-Gov-Logo-Gold.svg"
            width="46"
          />
        </a>
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            aria-expanded={null}
            className="navbar-collapse collapse"
          >
            <a
              className="nav-link"
              data-rb-event-key="https://edd.ca.gov"
              href="https://edd.ca.gov"
              onClick={[Function]}
              onKeyDown={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="text"
              >
                EDD Home
              </span>
            </a>
          </div>
          <a
            className="nav-link"
            data-rb-event-key="https://askedd.edd.ca.gov/"
            href="https://askedd.edd.ca.gov/"
            onClick={[Function]}
            onKeyDown={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="text"
            >
              Help
            </span>
          </a>
          <div
            aria-expanded={null}
            className="navbar-collapse collapse"
          >
            <a
              className="nav-link"
              data-rb-event-key="https://edd.ca.gov"
              href="https://edd.ca.gov"
              onClick={[Function]}
              onKeyDown={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="text"
              >
                Welcome, Firstname
              </span>
            </a>
          </div>
          <a
            className="nav-link"
            data-rb-event-key="https://edd.ca.gov/login.htm"
            href="https://edd.ca.gov/login.htm"
            onClick={[Function]}
            onKeyDown={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <span
              className="text"
            >
              Log out
            </span>
          </a>
        </div>
      </nav>
      <nav
        className="justify-content-between navbar navbar-expand-lg navbar-light"
      >
        <a
          className="navbar-brand"
          href="https://edd.ca.gov"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            className="d-inline-block align-top mr-5"
            height="60"
            src="/images/edd-logo-2-Color.svg"
            width="171"
          />
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <div
            className="navbar-nav"
            onKeyDown={[Function]}
          >
            <a
              className="nav-link"
              data-rb-event-key="https://uio.edd.ca.gov"
              href="https://uio.edd.ca.gov"
              onClick={[Function]}
              onKeyDown={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                className="text"
              >
                UI Online Home
              </span>
            </a>
          </div>
        </div>
      </nav>
    </header>
    <h1
      className="title"
    >
      Placeholder Claim Tracker Application
    </h1>
    <p />
    <button
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      En espa√±ol
    </button>
  </main>
  <footer
    className="footer"
  >
    Placeholder footer
  </footer>
</div>
`;
